### PR TITLE
HADOOP-16222. Fix new deprecations after guava 27.0 update in trunk

### PR DESCRIPTION
**Pull request by bgaborg**

### CODE REVIEW

**Comment by bgaborg** [comment]
Fix for `hadoop/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/SemaphoredDelegatingExecutor.java:[110,20] [deprecation] <V,X>immediateFailedCheckedFuture(X) in Futures has been deprecated` still missing, I'll add it in a follow-up commit because I need to think about it a little bit more. Just wanted to test if yetus is ok with this change meanwhile.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDFSHAAdminMiniCluster.java [hunk 1]
```diff
@@ -208,16 +208,16 @@ public void testFencer() throws Exception {
...
     // Test failover with fencer and forcefence option
     assertEquals(0, runTool("-failover", "nn1", "nn2", "--forcefence"));
     
     // The fence script should run with the configuration from the target
     // node, rather than the configuration from the fencing node. Strip
     // out any trailing spaces and CR/LFs which may be present on Windows.
-    String fenceCommandOutput =Files.toString(tmpFile, Charsets.UTF_8).
-            replaceAll(" *[\r\n]+", "");
+    String fenceCommandOutput = Files.asCharSource(tmpFile, Charsets.UTF_8).
+        read().replaceAll(" *[\r\n]+", "");
```
Comment (jiwq) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDFSHAAdminMiniCluster.java]: In chained method calls, if we need wrap it, I think the dot operator and method is best in the same line.

Comment (steveloughran) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDFSHAAdminMiniCluster.java]: +1: dot at the start. (note: on scala you should put it at the end so it doesn't think the line ends early, but this is .java here)

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 28 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 7 new or modified test files. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 23 | Maven dependency ordering for branch |
| -1 | mvninstall | 320 | root in trunk failed. |
| -1 | compile | 21 | root in trunk failed. |
| -0 | checkstyle | 6 | The patch fails to run checkstyle in root |
| -1 | mvnsite | 19 | hadoop-common in trunk failed. |
| -1 | mvnsite | 18 | hadoop-hdfs in trunk failed. |
| -1 | mvnsite | 14 | hadoop-yarn-server-nodemanager in trunk failed. |
| -1 | mvnsite | 19 | hadoop-yarn-server-resourcemanager in trunk failed. |
| -1 | mvnsite | 20 | hadoop-yarn-services-core in trunk failed. |
| +1 | shadedclient | 121 | branch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 24 | hadoop-common in trunk failed. |
| -1 | findbugs | 22 | hadoop-hdfs in trunk failed. |
| -1 | findbugs | 22 | hadoop-yarn-server-nodemanager in trunk failed. |
| -1 | findbugs | 112 | hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager in trunk has 2 extant Findbugs warnings. |
| +1 | javadoc | 168 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 22 | Maven dependency ordering for patch |
| +1 | mvninstall | 230 | the patch passed |
| +1 | compile | 996 | the patch passed |
| -1 | javac | 996 | root generated 1481 new + 0 unchanged - 0 fixed = 1481 total (was 0) |
| -0 | checkstyle | 192 | root: The patch generated 91 new + 0 unchanged - 0 fixed = 91 total (was 0) |
| +1 | mvnsite | 275 | the patch passed |
| -1 | whitespace | 0 | The patch has 61 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply |
| -1 | whitespace | 1 | The patch 600  line(s) with tabs. |
| +1 | shadedclient | 632 | patch has no errors when building and testing our client artifacts. |
| +1 | findbugs | 472 | the patch passed |
| +1 | javadoc | 210 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 508 | hadoop-common in the patch failed. |
| -1 | unit | 4723 | hadoop-hdfs in the patch failed. |
| +1 | unit | 1247 | hadoop-yarn-server-nodemanager in the patch passed. |
| +1 | unit | 4653 | hadoop-yarn-server-resourcemanager in the patch passed. |
| +1 | unit | 1080 | hadoop-yarn-services-core in the patch passed. |
| -1 | asflicense | 46 | The patch generated 1 ASF License warnings. |
| | | 16433 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.io.TestArrayWritable |
|   | hadoop.hdfs.TestReconstructStripedFileWithRandomECPolicy |
|   | hadoop.hdfs.server.datanode.TestBPOfferService |
|   | hadoop.hdfs.TestDFSStorageStateRecovery |
|   | hadoop.hdfs.TestDFSStripedOutputStream |
|   | hadoop.hdfs.web.TestWebHdfsTimeouts |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/692 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux e2af5e4bd605 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / dc013f7 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvninstall-root.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-compile-root.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out//testptch/patchprocess/maven-branch-checkstyle-root.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvnsite-hadoop-common-project_hadoop-common.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvnsite-hadoop-hdfs-project_hadoop-hdfs.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-mvnsite-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-applications_hadoop-yarn-services_hadoop-yarn-services-core.txt |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-findbugs-hadoop-common-project_hadoop-common.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-findbugs-hadoop-hdfs-project_hadoop-hdfs.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-nodemanager.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/branch-findbugs-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server_hadoop-yarn-server-resourcemanager-warnings.html |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/diff-compile-javac-root.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/diff-checkstyle-root.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/whitespace-eol.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/whitespace-tabs.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/testReport/ |
| asflicense | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/artifact/out/patch-asflicense-problems.txt |
| Max. process+thread count | 4957 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-hdfs-project/hadoop-hdfs hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-applications/hadoop-yarn-services/hadoop-yarn-services-core U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 12 | https://github.com/apache/hadoop/pull/692 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/692 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/6/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 12 | https://github.com/apache/hadoop/pull/692 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/692 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-692/7/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**